deepmd/pt/model/descriptor/env_mat.py

```diff
@@ -21,10 +21,11 @@ def _make_env_mat(
     nall = coord.shape[1]
     mask = nlist >= 0
     # nlist = nlist * mask  ## this impl will contribute nans in Hessian calculation.
-    nlist = torch.where(mask, nlist, nall - 1)
+    nlist = torch.where(mask, nlist, nall)
     coord_l = coord[:, :natoms].view(bsz, -1, 1, 3)
     index = nlist.view(bsz, -1).unsqueeze(-1).expand(-1, -1, 3)
-    coord_r = torch.gather(coord, 1, index)
+    coord_pad = torch.concat([coord, coord[:, -1:, :] + rcut], dim=1)
```

Comment:
Please @iProzd resolve this conversation.

Reply:
It's not necessary to consider this because:
1. Whatever `coord[:, -1:, :] + rcut` is, the computed weight will be masked using nlist `mask`.
2. If very small possibility `coord[:, -1:, :] + rcut` is the same as the coordinate of a real atom, the computed length will then be added a 1.0 protection to avoid being zeros, see `length = length + ~mask.unsqueeze(-1)` below.